common/djangoapps/edxmako/tests.py

```diff
@@ -63,6 +63,31 @@ def test_is_any_marketing_link_set(self):
             self.assertTrue(is_any_marketing_link_set(['ABOUT', 'NOT_CONFIGURED']))
             self.assertFalse(is_any_marketing_link_set(['NOT_CONFIGURED']))
 
+    @override_settings(MKTG_URLS={'ROOT': 'https://dummy-root', 'TOS': '/tos'})
+    @override_settings(MKTG_URL_OVERRIDES={'TOS': 'https://edx.org'})
+    def test_override_marketing_link(self):
+        expected_link = 'https://edx.org'
+        # test marketing site on
+        with patch.dict('django.conf.settings.FEATURES', {'ENABLE_MKTG_SITE': True}):
+            link = marketing_link('TOS')
+            self.assertEquals(link, expected_link)
+        # test marketing site off
+        with patch.dict('django.conf.settings.FEATURES', {'ENABLE_MKTG_SITE': False}):
+            link = marketing_link('TOS')
+            self.assertEquals(link, expected_link)
```

Comment:
If your check the code, it's just a override that changes the link if has any value. It takes precedence to avoid having to deal with all the changes related to enabling/disabling an external marketing website and allows us to keep using the internal marketing pages :grin:

Reply:
Oh yes, I understand that. I can't quite put my finger on what feels odd about it. I guess it just feels a bit like this mechanism should be *instead of* the marketing URLs, rather than overriding it. If we have `FEATURES['ENABLE_MKTG_SITE`]: false` then it seems like anything in `MKTG_URLS` is irrelevant because it shouldn't be included, so overriding it shouldn't make a difference.

But I think if we document this clearly it's probably fine.